### DISCUSSION TITLE
[RFC] Ensure certain scripts are implemented by all packages

### DISCUSSION THREAD

**Discussion by dmnd** [vercel/turborepo#7266] [Ideas]
### Describe the feature you'd like to request

I want to ensure that all packages in my monorepo implement a certain script.

### Describe the solution you'd like

- `yarn turbo run lint` is close, but it ignores packages that don't implement lint.
- `yarn turbo run --ensure lint` could fail if a package doesn't implement lint.

### Describe alternatives you've considered

`yarn workspaces run lint` runs lint in all packages, and fails if a package doesn't implement lint. But it's so slow ;)

**Comment by IPWright83** [comment]
@dmnd This sounds quite useful, though is Turborepo the place to do it? It feels a little bit like a "linting/pre-commit" hook type check?

I might actually see if I can knock a script up later, as I'd value this in our project.

**Comment by dmnd** [comment]
Maybe the turbo eslint plugin is a better home for it? Not sure though as this is kinda circular. In my monorepo lint is global so it'd work, but I bet others lint by package. 

Where would you put it?

**Comment by nahumzs** [comment]
Shouldn't this still require a folder filter or an array of package where you want to run `--ensure lint`? Otherwise, this implies that all packages must look the same, which in a lot of mono repo is not true unless you are running a UI-specific library why would you enforce all packages to have a `lint` script?

To me, this sounds more like a specific user behaviour rather than turbo repo responsibility.

**Comment by dmnd** [comment]
> why would you enforce all packages to have a `lint` script?

Monorepos are frequently used to increase consistency across projects. Also, lint is just an example.
